src/server/worker/pipeline/transform/cache.go

```diff
@@ -0,0 +1,70 @@
+package transform
+
+import (
+	"context"
+
+	"github.com/gogo/protobuf/types"
+	"github.com/pachyderm/pachyderm/v2/src/client"
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/pfs"
+)
+
+type cache struct {
+	pachClient *client.APIClient
+}
+
+func newCache(pachClient *client.APIClient) *cache {
+	return &cache{pachClient: pachClient}
+}
+
+func (c *cache) Get(ctx context.Context, key string) (*types.Any, error) {
+	resp, err := c.pachClient.PfsAPIClient.GetCache(ctx, &pfs.GetCacheRequest{Key: key})
+	if err != nil {
+		return nil, errors.EnsureStack(err)
+	}
+	return resp.Value, nil
+}
+
+func (c *cache) Put(ctx context.Context, key string, output *types.Any) error {
+	var fileSetIds []string
+	switch {
+	case types.Is(output, &UploadDatumsTaskResult{}):
+		udt, err := deserializeUploadDatumsTaskResult(output)
+		if err != nil {
+			return err
+		}
+		fileSetIds = append(fileSetIds, udt.FileSetId)
```

Comment:
Would that make things simpler if we still need the type switch here?

Reply:
I was imagining this function could turn into something like
```go
var msg types.DynamicAny
types.UnmarshalAny(output, &msg)
if lister, ok := msg.Message.(FileSetLister); ok {
    fileSetIds = append(fileSetIds, lister.List()...)
}
```
but regardless it doesn't need to be part of this PR

Reply:
IIUC this would be adding methods to the protocol buffers result types?  I think that's a good idea, we do something similar with the primitive and composite filesets.

It doesn't actually simplify the Put method does it, because of the Any?  Would we unwrap the Any and then check that it converts to the `FilesetTask` interface, and then error if it doesn't.  So in order to use the fileset cache the inner protocol buffer has to implement the interface.

Reply:
Yeah, I think that would just add complexity without any benefit at this point, but it would be a better design if we needed that information elsewhere. I don't think that is likely, but we could easily move to that model if that is ever the case.